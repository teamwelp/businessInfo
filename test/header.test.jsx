import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../client/dev/header';
import puppeteer from 'puppeteer';
import starterData from './starterdata';
import renderer from 'react-test-renderer';

const { shallow } = Enzyme;
Enzyme.configure({ adapter: new Adapter() });

describe('Header Component Unit Test', () => {
  test('renders a component', () => {
    const wrapper = shallow(<Header data={starterData} />);
    expect(wrapper.find('.header')).toHaveLength(1);
  });
  test('renders restaurant name from starter data', () => {
    const wrapper = shallow(<Header data={starterData} />);
    expect(wrapper.find('.name').text()).toBe('Oleg\'s Burger Palace');
  });
  test('renders restaurant claimedByOwner from starter data', () => {
    const wrapper = shallow(<Header data={starterData} />);
    expect(wrapper.find('.claimed').text()).toBe('Claimed');
  });
  test('renders restaurant metatags', () => {
    const wrapper = shallow(<Header data={starterData} />);
    expect(wrapper.find('.metatags').text()).toBe('Soul Food, Bars, Dinner, Brunch');
  });
});

describe('Header Component Snapshot Test', () => {
  test('renders correctly', () => {
    const newSnap = renderer
      .create(<Header data={starterData} />)
      .toJSON();
    expect(newSnap).toMatchSnapshot();
  });
})

describe('Header Component End-to-End Test', () => {
  test('rendering default page from server', async () => {
    const browser = await puppeteer.launch({
      headless: true,
    });
    const page = await browser.newPage();
    await page.goto('http://127.0.0.1:3000/');
    await page.waitForSelector('.header__name___uwB32');
    const html = await page.$eval('.header__name___uwB32', e => e.innerHTML);
    expect(html).toBe('Oleg\'s Burger Palace');
  }, 10000);
});
