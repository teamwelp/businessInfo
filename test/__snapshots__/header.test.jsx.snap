// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot Test renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="name"
    >
      Oleg's Burger Palace
       
      <div
        className="claimicon"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m31.7 16.4q0-0.6-0.4-1l-2.1-2.1q-0.4-0.4-1-0.4t-1 0.4l-9.1 9.1-5-5q-0.5-0.4-1-0.4t-1 0.4l-2.1 2q-0.4 0.4-0.4 1 0 0.6 0.4 1l8.1 8.1q0.4 0.4 1 0.4 0.6 0 1-0.4l12.2-12.1q0.4-0.4 0.4-1z m5.6 3.6q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
            />
          </g>
        </svg>
      </div>
      <div
        className="claimed"
      >
        Claimed
      </div>
    </div>
    <div
      className="bottomHeader"
    >
      $
      <div
        className="bullet"
      >
        •
      </div>
      <div
        className="metatags"
      >
        Soul Food, Bars, Dinner, Brunch
      </div>
    </div>
  </div>
  <div
    className="contact"
  >
    <div
      className="address"
    >
      <div
        className="grayicon"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m25.6 14.3q0-2.4-1.6-4.1t-4.1-1.6-4 1.6-1.7 4.1 1.7 4 4 1.7 4.1-1.7 1.6-4z m5.8 0q0 2.4-0.8 4l-8.1 17.3q-0.4 0.7-1.1 1.1t-1.5 0.4-1.5-0.4-1-1.1l-8.2-17.3q-0.7-1.6-0.7-4 0-4.7 3.3-8.1t8.1-3.3 8.1 3.3 3.4 8.1z"
            />
          </g>
        </svg>
      </div>
      <div
        className="addressPhone"
      >
        301
         
        Mission Street
        <br />
        San Francisco
        , 
        CA
         
        94103
      </div>
    </div>
    <div>
      <div
        className="grayicon"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m35.9 27.7q0 0.6-0.2 1.6t-0.5 1.5q-0.4 1.1-2.7 2.3-2.1 1.2-4.1 1.2-0.6 0-1.2-0.1t-1.3-0.3-1.1-0.3-1.2-0.5-1.1-0.4q-2.2-0.7-3.9-1.8-2.8-1.8-5.9-4.8t-4.8-5.9q-1.1-1.7-1.9-3.9 0-0.2-0.4-1.1t-0.4-1.3-0.3-1-0.3-1.3-0.1-1.2q0-2 1.1-4.1 1.3-2.3 2.4-2.8 0.6-0.2 1.5-0.4t1.6-0.2q0.3 0 0.5 0 0.4 0.2 1.2 1.7 0.2 0.4 0.6 1.2t0.8 1.4 0.7 1.2q0.1 0.1 0.4 0.6t0.5 0.8 0.1 0.6q0 0.5-0.6 1.1t-1.4 1.3-1.4 1.1-0.6 1.1q0 0.2 0.1 0.5t0.2 0.4 0.3 0.6 0.3 0.4q1.7 3.1 3.8 5.2t5.3 3.9q0 0 0.4 0.3t0.6 0.3 0.4 0.2 0.5 0.1q0.4 0 1-0.6t1.2-1.4 1.3-1.4 1.1-0.6q0.3 0 0.6 0.1t0.8 0.5 0.6 0.4q0.5 0.3 1.1 0.7t1.5 0.8 1.2 0.6q1.5 0.8 1.7 1.2 0 0.2 0 0.5z"
            />
          </g>
        </svg>
      </div>
      <div
        className="addressPhone"
      >
        (
        415
        ) 
        421
        -
        7940
      </div>
    </div>
    <div>
      <div
        className="grayicon"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m31.4 20.7v7.2q0 2.6-1.9 4.5t-4.5 1.9h-18.6q-2.6 0-4.5-1.9t-1.9-4.5v-18.6q0-2.7 1.9-4.6t4.5-1.8h15.7q0.4 0 0.6 0.2t0.2 0.5v1.4q0 0.3-0.2 0.5t-0.6 0.2h-15.7q-1.4 0-2.5 1.1t-1 2.5v18.6q0 1.4 1 2.5t2.5 1h18.6q1.5 0 2.5-1t1.1-2.5v-7.2q0-0.3 0.2-0.5t0.5-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m8.6-19.3v11.5q0 0.5-0.4 1t-1 0.4-1-0.4l-4-4-14.5 14.6q-0.2 0.2-0.5 0.2t-0.5-0.2l-2.6-2.6q-0.2-0.2-0.2-0.5t0.2-0.5l14.6-14.5-4-4q-0.4-0.4-0.4-1t0.4-1 1-0.4h11.5q0.6 0 1 0.4t0.4 1z"
            />
          </g>
        </svg>
      </div>
      <a
        className="contactLink"
        href="http://olegsburgerpalace.com"
      >
        http://olegsburgerpalace.com
      </a>
    </div>
    <div>
      <div
        className="grayicon"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m21.9 31.4q0-0.7-0.6-1.2t-1.2-0.6-1.3 0.6-0.5 1.2 0.5 1.3 1.3 0.5 1.2-0.5 0.6-1.3z m4.6-3.5v-15.8q0-0.2-0.2-0.5t-0.5-0.2h-11.4q-0.3 0-0.5 0.2t-0.3 0.5v15.8q0 0.2 0.3 0.5t0.5 0.2h11.4q0.3 0 0.5-0.2t0.2-0.5z m-4.3-19q0-0.3-0.3-0.3h-3.6q-0.4 0-0.4 0.3t0.4 0.4h3.6q0.3 0 0.3-0.4z m6.4-0.3v22.8q0 1.2-0.8 2t-2 0.9h-11.4q-1.2 0-2.1-0.9t-0.8-2v-22.8q0-1.2 0.8-2t2.1-0.9h11.4q1.1 0 2 0.9t0.8 2z"
            />
          </g>
        </svg>
      </div>
      <a
        className="contactLink"
        href="#"
      >
        Send to your Phone
      </a>
    </div>
    <div
      className="clearBoth"
    />
  </div>
</div>
`;
